### ISSUE TITLE
submitPayment returns error "Invalid scope"

### ISSUE THREAD

**Issue by Perjan** [figo-connect/ios-sdk#11]
Hello guys,
I am having an issue submitting a payment. I am trying to submit a **demo payment** with the **demo account** provided by you.

Steps:
1. I call `createPayment(_ parameters: CreatePaymentParameters, _ completionHandler: @escaping (FigoResult<Payment>) -> Void)`
2. I get a `Payment` object back
3. I pass that object and the tanSchemeID = "111111"  to `submitPayment(_ payment: Payment, tanSchemeID: String, pinHandler: @escaping PinResponder, challengeHandler: @escaping ChallengeResponder, _ completionHandler: @escaping VoidCompletionHandler) `

After that I get the error "Invalid scope" with the code "90000"

What am I doing wrong?
Maybe the tanSchemeID is wrong?

Thanks in advance.

**Comment by CodeStage** [comment]
Hi @Perjan,
the tanSchemeID should look something like "M2132899.9". You have to get the ID from the supportedTANSchemes property of your account.

**Comment by Perjan** [comment]
Thanks Christian, will try it out and let you know.

Also the Challenge closure is not that well documented.
A little help on that one.

**Comment by CodeStage** [comment]
>  A closure which is called when the server needs a response to a challenge from the user
> 
>  - Parameter message: Status message or error message for currently processed account
>  - Parameter accountID: Account ID of currently processed account
>  - Parameter challenge: Challenge object

> public typealias ChallengeResponder = (_ message: String, _ accountID: String, _ challenge: Challenge) -> String


You can check out the test cases, to see how it works. For example testCreateModifyAndSubmitPayment() hast the challenge closure.

**Comment by Perjan** [comment]
Thanks a lot @CodeStage, I will check them out today and tomorrow and let you know.
Have a great weekend!

**Comment by Perjan** [comment]
Hey @CodeStage I changed my method today, created a new ChallengeResponder object, passing the accountID of a demo GiroAccount, and passing the tanSchemeID of an iTAN.

Now I can create a payment successfully, but when I submit it, I still get the "Invalid Scope" error.
Could it be that I am using the demo account?

Please help me resolve this.
Thanks in advance,
P

**Comment by CodeStage** [comment]
@Perjan Did you look at the "testCreateModifyAndSubmitPayment" test case?

**Comment by Perjan** [comment]
@CodeStage yes I did, and I took the ChallengeResponder and the PinHandler objects from it, pasted them into my method, but still I am getting an invalid scope.

Could it be that the demo account provided by figo cannot do payments?

Thanks,
P

**Comment by Perjan** [comment]
@CodeStage Can you please help me out with this issue?

Thanks a lot,
P

**Comment by CodeStage** [comment]
@Perjan I'm afraid I can't help you. I verified that the test case is still working and it is. It uses the demo account to create and submit a payment. So you should be able to setup the demo account with your user and after downloading the account object you should be able to create and submit payments with the IDs from that object.

**Comment by Perjan** [comment]
I am using also the demo account. I can create a payment, and I get a successful response. The issue is with the submitPayment method, I get a failure out of that...

That is why I thought maybe there is an issue withe API keys that I am using.

**Comment by CodeStage** [comment]
Yeah, maybe you should try to rule that out by asking support@figo.io if you are using the right keys. Plus they might be able to tell you in which case they are sending you the "Invalid Scope" error.

**Comment by Perjan** [comment]
Awesome, thanks for helping me out.
Will talk to the support guys.

Cheers,
P
